### PR TITLE
feat(unstable): add Deno.futime and Deno.futimeSync

### LINKED ISSUE #6563: Add support for futimes
Add support for futimes(2) to allow changing the timestamps of opened file

```typescript
namespace Deno {
  export function futimesSync(rid: number, atime: number, mtime: number): void;
  export function futimes(rid: number, atime: number, mtime: number): Promise<void>;
}

### PR DESCRIPTION
**Pull request by caspervonb**
This adds Deno.futime and Deno.futimeSync with the same semantics as Deno.utime and Deno.utimeSync operating on file resource streams rather than file system paths.

Closes #6563

### CODE REVIEW

**Comment by caspervonb** [comment]
I'll follow up with a patch to move replace the use of the utime crate in utime ops with the the filetime crate.

**Comment by caspervonb** [comment]
PTAL @ry

File: cli/Cargo.toml [hunk 1]
```diff
@@ -42,6 +42,7 @@ dlopen = "0.1.8"
 encoding_rs = "0.8.23"
 dprint-plugin-typescript = "0.30.1"
 futures = "0.3.5"
+filetime = "0.2.12"
```
Comment (ry) [inline cli/Cargo.toml:45]: Is this necessary?

Comment (caspervonb) [reply cli/Cargo.toml:45]: We can do it by libc and winapi I suppose but I'm satisfied with the implementation (uses futimens) in filetime and it makes it a readable one liner.

Comment (caspervonb) [reply cli/Cargo.toml:45]: Oh and the utime crate we currently have a dependency on doesn't provide any fd/handle based functions, just the one path based one.

So to clarify; the choices are filetime crate or write libc code duplicating what the filetime crate does.

Comment (ry) [reply cli/Cargo.toml:45]: Fair enough. I will land this as is then.

**Pull request merged**